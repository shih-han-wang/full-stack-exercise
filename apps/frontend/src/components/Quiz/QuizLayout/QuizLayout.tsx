import { Flex, Heading } from '@chakra-ui/react'
import { ReactElement } from 'react'

type Props = {
  children: ReactElement | ReactElement[]
  heading: string
}

const QuizLayout = ({ children, heading }: Props) => (
  <Flex
    height='100vh'
    alignItems='center'
    justifyContent='center'
    flexDirection='column'
  >
    <Heading as='h1' size='2xl' mb={8} textAlign='center'>
      {heading}
    </Heading>

    {children}
  </Flex>
)

export default QuizLayout
