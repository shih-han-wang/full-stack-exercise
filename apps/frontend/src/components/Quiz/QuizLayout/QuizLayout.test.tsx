import { render } from '@testing-library/react'
import QuizLayout from './QuizLayout'

describe('Components > Quiz > QuizLayout', () => {
  it('should render correctly', async () => {
    const view = render(
      <QuizLayout heading='Ask me a question.'>
        <div>children</div>
      </QuizLayout>
    )

    expect(view.asFragment()).toMatchSnapshot()
  })
})
