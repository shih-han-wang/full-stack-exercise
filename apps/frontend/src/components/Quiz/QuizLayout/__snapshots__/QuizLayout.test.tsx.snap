// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Quiz > QuizLayout should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <h1
      class="chakra-heading css-0"
    >
      Ask me a question.
    </h1>
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;
